pkg/cloud/gen/main.go

```diff
@@ -728,15 +728,33 @@ func (g *{{.GCEWrapType}}) List(ctx context.Context, zone string, fl *filter.F)
 		call.Filter(fl.String())
 	}
 	var all []*{{.FQObjectType}}
+  var nextPageToken string
 	f := func(l *{{.ObjectListType}}) error {
 		klog.V(5).Infof("{{.GCEWrapType}}.List(%v, ..., %v): page %+v", ctx, fl, l)
+		nextPageToken = l.NextPageToken
 		all = append(all, l.Items...)
 		return nil
 	}
-	if err := call.Pages(ctx, f); err != nil {
-		klog.V(4).Infof("{{.GCEWrapType}}.List(%v, ..., %v) = %v, %v", ctx, fl, nil, err)
-		return nil, err
-	}
+	
+  var retryCount int
+  for {
+    select {
+		case <-ctx.Done():
+			klog.V(5).Infof("{{.GCEWrapType}} call.Pages(%v, _) not completed, poll count = %d, ctx.Err = %v", ctx, retryCount, ctx.Err())
+			return nil, ctx.Err()
+		default:
+			// ctx is not canceled, continue immediately
+		}
+		if nextPageToken != "" {
+			call.PageToken(nextPageToken)
+		}
+		if err := call.Pages(ctx, f); err != nil {
```

Comment:
Are there different kinds of errors we want to handle differently?

Reply:
I don't think so, are there any specific response codes you think we should handle?  We don't handle response codes individually in the other functions AFAIK

Reply:
@bowei @spencerhance please let me know if you need any help with this! We're seeing some flakes in our CI that I think would be solved with this patch.